plugins/media-keys/gsd-media-keys-manager.c

```diff
@@ -3144,6 +3152,40 @@ power_keyboard_ready_cb (GObject             *source_object,
         }
 }
 
+static gboolean
+lid_closed_set (GsdMediaKeysManager *manager)
+{
+        g_debug ("lid is now %s", manager->priv->lid_is_closed_tmp ? "closed" : "open");
+        manager->priv->lid_is_closed = manager->priv->lid_is_closed_tmp;
+        manager->priv->inhibit_tp_toggle_timer_id = 0;
+        return G_SOURCE_REMOVE;
+}
+
+static void
+lid_state_changed_cb (UpClient *client, GParamSpec *pspec, GsdMediaKeysManager *manager)
+{
+        if (!manager->priv->lid_is_present)
+                return;
+
+        manager->priv->lid_is_closed_tmp = up_client_get_lid_is_closed (manager->priv->up_client);
```

Comment:
I think we should trust the property notification to be emitted when the value actually changes. But in reality, even if it doesn't there's no advantage to returning early here that I can see.